src/dynamics/periodic_pts.lean

```diff
@@ -302,6 +310,38 @@ begin
   exact nat.dvd_lcm_left _ _ <|> exact dvd_lcm_right _ _
 end
 
+lemma commute.minimal_period_of_comp_dvd_mul_of_coprime {g : α → α} (h : function.commute f g)
+  (hco : coprime (minimal_period f x) (minimal_period g x)) :
+  minimal_period (f ∘ g) x ∣ (minimal_period f x) * (minimal_period g x) :=
+begin
+  rw ←hco.lcm_eq_mul,
+  exact h.minimal_period_of_comp_dvd_lcm
+end
+
+lemma commute.minimal_period_of_comp_eq_mul_of_coprime {g : α → α} (h : function.commute f g)
+  (hco : coprime (minimal_period f x) (minimal_period g x)) :
+  minimal_period (f ∘ g) x = (minimal_period f x) * (minimal_period g x) :=
+begin
+  apply dvd_antisymm (h.minimal_period_of_comp_dvd_mul_of_coprime hco),
+  apply coprime.mul_dvd_of_dvd_of_dvd hco,
```

Comment:
You can write the same proof shorter if you use dot notation.

Reply:
I changed, but maybe you had some more golfing in mind?